server/graphql/v2/mutation/CommentReactionMutations.js

```diff
@@ -0,0 +1,51 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import models from '../../../models';
+import { Unauthorized } from '../../errors';
+import { CommentReactionCreateInput } from '../input/CommentReactionCreateInput';
+import { CommentReaction } from '../object/CommentReaction';
+
+async function createCommentReaction(entity, args, req) {
+  if (!req.remoteUser) {
+    throw new Unauthorized();
+  }
+
+  const { emoji, comment } = args;
+
+  return await models.CommentReaction.addCommentReaction(req.remoteUser, comment, emoji);
+}
+
+async function deleteCommentReaction(id, remoteUser) {
+  if (!remoteUser) {
+    throw new Unauthorized();
+  }
+  return await models.CommentReaction.addCommentReaction(id);
+}
+
+const commentReactionMutations = {
+  addCommentReaction: {
+    type: CommentReaction,
+    args: {
+      commentReaction: {
+        type: new GraphQLNonNull(CommentReactionCreateInput),
+      },
+    },
+    resolve: async (entity, args, req) => {
+      return createCommentReaction(entity, args, req);
+    },
+  },
+
+  removeCommentReaction: {
+    type: CommentReaction,
+    args: {
+      id: {
+        type: new GraphQLNonNull(GraphQLString),
+      },
+    },
+    resolve(_, { id }, { remoteUser }) {
+      return deleteCommentReaction(id, remoteUser);
```

Comment:
According to the code below, this should be `removeReaction`. We also need to check the permissions before removing the reaction

Reply:
Thanks much for the review. I've added the code to check the permissions. Let me know if you see any further issues. :smiley: